### PR TITLE
[WIP] Add enable_osr flag

### PR DESCRIPTION
**Pull request by alexeykuzmin**
It would be great to have an ability to turn certain features off for custom builds.
It could also help with upgrades to new Chromium versions.

**Issue to address**:
Disable the tests if the flag is turned off.

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
All Travis builds fail on downloading libcontentchromium.zip archive.
@zcbenz, any chance you know what could be wrong?

**Comment by zcbenz** [comment]
@alexeykuzmin Our Linux CI machines require GitHub employee's permission to run, I have manually triggered the builds, which will take a while to run.

File: common.gypi [hunk 1]
```diff
@@ -246,6 +247,13 @@
       4838,  # (atlgdi.h) conversion from 'int' to 'UINT' requires a narrowing conversion
       4996,  # (atlapp.h) 'GetVersionExW': was declared deprecated
     ],
+    'conditions': [
+      ['enable_osr==1', {
```
Comment (zcbenz) [inline common.gypi]: The `common.gypi` is a place to put settings for dependencies like Node and breakpad, for configurations of Electron itself, it is probably better to put it in `electron.gyp`.

Comment (alexeykuzmin) [inline common.gypi]: Done.

**Comment by zcbenz** [comment]
For the tests on offscreen, I think you can detect whether offscreen is enabled by creating an offscreen WebContents and then testing its `isOffScreen()` method.

**Comment by alexeykuzmin** [comment]
> For the tests on offscreen, I think you can detect whether offscreen is enabled by creating an offscreen WebContents and then testing its `isOffScreen()` method.

Ok. I'll try to do that. Thanks.

**Comment by alexeykuzmin** [comment]
> For the tests on offscreen, I think you can detect whether offscreen is enabled by creating an offscreen WebContents and then testing its `isOffScreen()` method.

I've added skipping of the OSR tests if 'isOffscreen()' method of a WebContents instance doesn't exist.
For me absence of the `isOffscreen()` method looks logical when OSR feature is disabled at build time.

**Comment by kevinsawicki** [comment]
Thanks for this @alexeykuzmin 👍 

I pushed a small tweak to the spec and reverted the `script/update.py` change that disabled OSR on this branch. I'll merge this master into the 59 branch after merging this and disable OSR there.

**Comment by alexeykuzmin** [comment]
@kevinsawicki Have you tried to run tests on this branch? I have seen weird test failures on build bots in this PR. There is a chance that it was a glitch and everything is ok, but I'm not sure.

**Comment by kevinsawicki** [comment]
> Have you tried to run tests on this branch?

That may have been related to the spec change I pushed, you had it previously using `.skip()` which can cause issues with after/afterEach blocks not executing (and windows being leaked because of that) so we typically don't use it.

**Comment by alexeykuzmin** [comment]
> windows being leaked because of that

That was probably the reason why those tests failed. Cool.

**Pull request merged**